### PR TITLE
Reorg Bayesian optimizers to a single class per file

### PR DESCRIPTION
**Pull request by bpkroth**
Addresses a comment in #341

### CODE REVIEW

File: mlos_core/mlos_core/optimizers/bayesian_optimizers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+from mlos_core.optimizers.bayesian_optimizers.bayesian_optimizer import BaseBayesianOptimizer
+from mlos_core.optimizers.bayesian_optimizers.emukit_optimizer import EmukitOptimizer
+from mlos_core.optimizers.bayesian_optimizers.skopt_optimizer import SkoptOptimizer
+
+
+__all__ = [
+    'BaseBayesianOptimizer',
+    'EmukitOptimizer',
+    'SkoptOptimizer',
```
Comment (bpkroth) [inline mlos_core/mlos_core/optimizers/bayesian_optimizers/__init__.py:17]: - [ ] May want to add test cases that all subclasses of `BaseOptimizer` class are available in the factory methods, enum classes, etc.
  (this is relevant in the config test automation for `mlos_bench` as well - #340)

File: mlos_bench/mlos_bench/launcher.py
Comment (bpkroth) [inline mlos_bench/mlos_bench/launcher.py:1]: Not related - just applying the same `mypy` `# type: ignore[type-abstract]` tweak as done in the `get_all_concrete_subclasses()`

File: mlos_core/mlos_core/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+T = TypeVar('T')
+
+
+def _get_all_subclasses(cls: Type[T]) -> Set[Type[T]]:
+    """
+    Gets the set of all of the subclasses of the given class.
+    Useful for dynamically enumerating expected test cases.
+    """
+    return set(cls.__subclasses__()).union(
+        [s for c in cls.__subclasses__() for s in _get_all_subclasses(c)])
```
Comment (motus) [inline mlos_core/mlos_core/tests/__init__.py:21]: no need to make it a list - keep it as a generator (i.e., remove the `[]` around the expression)

File: mlos_core/mlos_core/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    Gets a sorted list of all of the concrete subclasses of the given class.
+    Useful for dynamically enumerating expected test cases.
+
+    Note: For abstract types, mypy will complain at the call site.
+    Use "# type: ignore[type-abstract]" to suppress the warning.
+    See Also: https://github.com/python/mypy/issues/4717
+    """
+    return sorted([subclass for subclass in _get_all_subclasses(cls) if not getattr(subclass, "__abstractmethods__", None)],
+                  key=lambda x: x.__module__ + "." + x.__name__)
```
Comment (motus) [inline mlos_core/mlos_core/tests/__init__.py:34]: maybe just
```suggestion
                  key=str)
```
or, more explicitly,
```suggestion
                  key=lambda cls: (cls.__module__, cls.__name__))
```

**Review by motus — APPROVED** [review]
left a few nitpicking comments; otherwise LGTM

**Pull request merged**